socorro/signature/signature_utilities.py

```diff
@@ -368,121 +373,84 @@ def _do_generate(self, source, hang_type_unused=0, crashed_thread_unused=None, d
                 (java_exception_class, description_java_method_phrase)
             )
 
-        if len(signature) > self.max_len:
+        if len(signature) > SIGNATURE_MAX_LENGTH:
             signature = delimiter.join(
                 (java_exception_class, java_method)
             )
             signature_notes.append(
-                'JavaSignatureTool: dropped Java exception '
-                'description due to length'
+                'JavaSignatureTool: dropped Java exception description due to length'
             )
 
         return signature, signature_notes
 
 
 class SignatureGenerationRule(Rule):
-    required_config = Namespace()
-    required_config.namespace('c_signature')
-    required_config.c_signature.add_option(
-        'c_signature_tool_class',
-        doc='the class that can generate a C signature',
-        default='socorro.processor.signature_utilities.CSignatureTool',
-        from_string_converter=class_converter,
-        reference_value_from='resource.signature'
-    )
-    required_config.c_signature.add_option(
-        'maximum_frames_to_consider',
-        doc='the maximum number of frames to consider',
-        default=40,
-        reference_value_from='resource.signature'
-    )
-    required_config.namespace('java_signature')
-    required_config.java_signature.add_option(
-        'java_signature_tool_class',
-        doc='the class that can generate a Java signature',
-        default='socorro.processor.signature_utilities.JavaSignatureTool',
-        from_string_converter=class_converter,
-        reference_value_from='resource.signature'
-    )
 
-    def __init__(self, config):
-        super(SignatureGenerationRule, self).__init__(config)
-        self.java_signature_tool = (
-            self.config.java_signature.java_signature_tool_class(
-                config.java_signature
-            )
-        )
-        self.c_signature_tool = self.config.c_signature.c_signature_tool_class(
-            config.c_signature
-        )
+    def __init__(self):
+        super(SignatureGenerationRule, self).__init__()
+        self.java_signature_tool = JavaSignatureTool()
+        self.c_signature_tool = CSignatureTool()
```

Comment:
The `java_signature_tool` and `c_signature_tool` used to be configurable. I changed it so that the classes are hard-coded now.